pkg/streamingpromql/operators/aggregations/quantile.go

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/engine.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors
+
+package aggregations
+
+import (
+	"math"
+	"unsafe"
+
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/parser/posrange"
+	"github.com/prometheus/prometheus/util/annotations"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/operators/functions"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+	"github.com/grafana/mimir/pkg/util/pool"
+)
+
+type QuantileAggregationGroup struct {
+	heaps []heap // A heap per point in time
+}
+
+type heap struct {
```

Comment:
[nit] Is there another name we could use for this? This isn't implementing a heap, it's just a list of values.

Alternatively: should this be a heap? `functions.Quantile()` sorts the entire list, but that's overkill, we just need a way to get the first X values.

Reply:
This is the terminology used by prometheus, but I agree it's not strictly a heap. I'll rename to `qGroup` since `group` is taken by the aggregation.